pkg/securitypolicy/framework.rego

```diff
@@ -3,31 +3,66 @@ package framework
 import future.keywords.every
 import future.keywords.in
 
-default mount_device := false
-mount_device := true {
+device_mounted(target) {
+    data.metadata.devices[target]
+}
+
+default mount_device := {"allowed": false}
+
+mount_device := {"devices": devices, "allowed": true} {
+    not device_mounted(input.target)
     some container in data.policy.containers
     some layer in container.layers
     input.deviceHash == layer
+    devices := {
+        "action": "add",
+        "key": input.target,
+        "value": input.deviceHash
+    }
 }
 
-layerPaths_ok(container) {
-    length := count(container.layers)
+default unmount_device := {"allowed": false}
+
+unmount_device := {"devices": devices, "allowed": true} {
+    device_mounted(input.unmountTarget)
+    devices := {
+        "action": "remove",
+        "key": input.unmountTarget
+    }
+}
+
+layerPaths_ok(layers) {
+    length := count(layers)
     count(input.layerPaths) == length
     every i, path in input.layerPaths {
-        container.layers[length - i - 1] == data.devices[path]
+        layers[(length - i) - 1] == data.metadata.devices[path]
     }
 }
 
-default mount_overlay := false
-mount_overlay := true {
-    some container in data.policy.containers
-    layerPaths_ok(container)
+default overlay_exists := false
+
+overlay_exists := {data.metadata.matches[input.containerID]}
```

Comment:
Is this an "inline" version of, e.g., `container_started` on L90? Would you prefer one over the other for consistency? Or will this evaluate only once to `false`?

Reply:
The two you mention are equivalent, but I've changed this to match `container_started` for consistency.